rclpy/rclpy/impl/rcutils_logger.py

```diff
@@ -215,15 +215,31 @@ class RcutilsLogger:
 
     def __init__(self, name=''):
         self.name = name
+        self.logger_sublogger_namepair = None
         self.contexts = {}
 
+    def __del__(self):
+        if self.logger_sublogger_namepair:
+            _rclpy.rclpy_logging_rosout_remove_sublogger(
+                self.logger_sublogger_namepair[0], self.logger_sublogger_namepair[1])
+            self.logger_sublogger_namepair = None
+
     def get_child(self, name):
         if not name:
             raise ValueError('Child logger name must not be empty.')
+
+        add_sublogger_flag = False
         if self.name:
             # Prepend the name of this logger
-            name = self.name + '.' + name
-        return RcutilsLogger(name=name)
+            add_sublogger_flag = _rclpy.rclpy_logging_rosout_add_sublogger(self.name, name)
```

Comment:
this comes false means that the subordinate logger already exists in hash map, that actually means that somebody else in the same process space already owns it, so just returning that child logger object without relationship saved in `logger_sublogger_namepair`? right?

Reply:
Yes.